operating_unit/models/res_users.py

```diff
@@ -0,0 +1,33 @@
+# © 2015-2017 Eficent
+# - Jordi Ballester Alomar
+# © 2015-2017 Serpent Consulting Services Pvt. Ltd. - Sudhir Arya
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+from odoo import api, fields, models
+
+
+class ResUsers(models.Model):
+
+    _inherit = 'res.users'
+
+    @api.model
+    def operating_unit_default_get(self, uid2):
+        if not uid2:
+            uid2 = self._uid
+        user = self.env['res.users'].browse(uid2)
+        return user.default_operating_unit_id
+
+    @api.model
+    def _get_operating_unit(self):
+        return self.operating_unit_default_get(self._uid)
```

Comment:
@simahawk Honestly I don't know, not the original author and I just couldn't understand why it was like it was.  Plus I've not looked at this in months. But it is different, passing nothing would give an error.  

I think I changed most of this file, then reverted it because other modules used this code.  

But in practice I've found these methods are unsuitable anyway and have rewritten because the only time you really care about a users default operating unit is in creation of a business document, and then you get all sorts of weird issues (like in a sales order, the warehouse,  Sales Team, salesperson and the user entering might all belong to different OU's and in multicompany it is a nightmare).

Reply:
@gdgellatly we are already doing it https://github.com/OCA/operating-unit/blob/10.0/sale_operating_unit/models/sale.py#L14
This is implementation is really old, there's no reason to keep it as it is. 
The ones that rely on this old way will have to migrate modules to v11 as well. 
So, this is the right time to make this change ;)

Same goes for the indirection of default methods: ppl used to add 2 methods to make them overridable but if you use lambda to proxy the call you don't need the double step anymore.